src/App.tsx

```diff
@@ -15,48 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortOptions {
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+  Reset = 'reset',
+}
+
+const handleSort
+  = (goods: string[],
+    options: { activeSort: SortOptions; isReversed: boolean }) => {
+    const preparedGoods = [...goods];
+
+    preparedGoods.sort((a, b) => {
+      switch (options.activeSort) {
+        case SortOptions.Alphabetically:
+          return a.localeCompare(b);
+
+        case SortOptions.Length:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (options.isReversed) {
+      preparedGoods.reverse();
+    }
+
+    return preparedGoods;
+  };
+
 export const App: React.FC = () => {
+  const [activeSort, setActiveSort]
+    = useState('');
+
+  const [isReversed, setIsReversed]
+    = useState(false);
+
+  const visibleGoods = handleSort(goodsFromServer, {
+    activeSort: activeSort || SortOptions,
+    isReversed,
+  } as { activeSort: SortOptions; isReversed: boolean });
+
+  function setAlphabetically() {
+    setActiveSort('length');
+  }
+
+  function setLength() {
+    setActiveSort('length');
+  }
```

Comment:
```suggestion
  function setSort(sort: SortOpion) {
     return () => setActiveSort(sort);
  }
```
sorting works little wrong. suggest to make one function for both sort buttons and call it with proper SortOptions value in jsx
![chrome-capture-2023-10-17](https://github.com/mate-academy/react_list-of-goods-hooks/assets/33029864/6912553b-fec2-4f15-8a5a-2750adf079d5)